src/scripts/migrate.py

```diff
@@ -52,10 +52,37 @@
     'mondo_id': 'ID', 'mondo_label': 'LABEL', 'xref': 'A oboInOwl:hasDbXref',
     'xref_source': '>A oboInOwl:source SPLIT=|', 'original_label': '', 'definition': 'A IAO:0000115', 'parents': 'SC %'}
 
+def _check_parent_conditions(t: Term, sssom_object_ids: Set[Union[URI, CURIE]], sssom_df: pd.DataFrame) -> bool:
+    """Determines if the term is migratable to Mondo
+
+    To be migratable, the term (i) must not already be mapped, (ii) must not be excluded (e.g. not in
+    `reports/%_term_exclusions.txt`), and (iii) must not be deprecated / obsolete. Then, (iv) there are conditions
+    concerning its parents. It must either (a) have no parents, or (b) have no valid parents in Mondo (i.e. all of its
+    parent terms are marked obsolete in Mondo), or (c) all its parents must be mapped, and at least 1 of those
+    parent's mappings must be one of `skos:exactMatch` or `skos:NarrowMatch`."""
+    obsolete_mondo_parent_ids = []
+    qualified_parent_mondo_ids = []
+    all_parents_mapped = True
+    no_parents: bool = not t.direct_owned_parent_curies
+    for parent_curie in t.direct_owned_parent_curies:
+        if parent_curie not in sssom_object_ids:
+            all_parents_mapped = False
+        if parent_curie in sssom_object_ids:
+            parent: Dict = sssom_df[sssom_df['object_id'] == parent_curie].to_dict('records')[0]
+            parent_mondo_id, parent_mondo_label = parent['subject_id'], parent['subject_label']
+            if parent_mondo_label.startswith('obsolete'):
+                obsolete_mondo_parent_ids.append(parent_mondo_id)
+            elif parent['predicate_id'] in ['skos:exactMatch', 'skos:narrowMatch']:
```

Comment:
I think this is mixing two concerns:

1. The condition under which a record should be declared mappable/added to the migration table (not obsolete, not excluded, all parents either mapped, obsoleted or excluded) and
2. What should be included in the "parents" column (of the parent is either an exact or narrow match). 

(2) is _not_ a further condition for adding the term to the slurp.

Reply:
I'm really glad you clarified that, thank you! Will do this now.